modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBMessageServiceImpl.java

```diff
@@ -101,10 +103,12 @@ public MBMessage addDiscussionMessage(
 
 		User user = getGuestOrUser();
 
-		MBDiscussionPermission.check(
-			getPermissionChecker(), user.getCompanyId(),
-			serviceContext.getScopeGroupId(), className, classPK,
-			ActionKeys.ADD_DISCUSSION);
+		DiscussionPermission discussionPermission =
+			_commentManager.getDiscussionPermission(getPermissionChecker());
+
+		discussionPermission.checkAddPermission(
+			user.getCompanyId(), serviceContext.getScopeGroupId(), className,
+			classPK);
 
```

Comment:
@W2jx, the changed logic is different from the original logic. For the original logic, when throw PrincipalException.MustHavePermission(), the userId is from permissionChecker.getUserId(). After the commit, the userId is 0.